src/main/java/de/tum/in/www1/artemis/service/connectors/AtheneService.java

```diff
@@ -0,0 +1,269 @@
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.Language;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.repository.TextBlockRepository;
+import de.tum.in.www1.artemis.repository.TextClusterRepository;
+import de.tum.in.www1.artemis.repository.TextExerciseRepository;
+import de.tum.in.www1.artemis.service.TextAssessmentQueueService;
+import de.tum.in.www1.artemis.service.TextBlockService;
+import de.tum.in.www1.artemis.service.TextSubmissionService;
+import de.tum.in.www1.artemis.web.rest.dto.AtheneDTO;
+
+@Service
+@Profile("athene")
+public class AtheneService {
+
+    private final Logger log = LoggerFactory.getLogger(AtheneService.class);
+
+    @Value("${server.url}")
+    protected String ARTEMIS_SERVER_URL;
+
+    @Value("${artemis.athene.submit-url}")
+    private String API_ENDPOINT;
+
+    @Value("${artemis.athene.base64-secret}")
+    private String API_SECRET;
+
+    private final TextAssessmentQueueService textAssessmentQueueService;
+
+    private final TextBlockRepository textBlockRepository;
+
+    private final TextBlockService textBlockService;
+
+    private final TextClusterRepository textClusterRepository;
+
+    private final TextExerciseRepository textExerciseRepository;
+
+    private final TextSubmissionService textSubmissionService;
+
+    private final RemoteArtemisServiceConnector<Request, Response> connector = new RemoteArtemisServiceConnector<>(log, Response.class);
+
+    // Contains tasks submitted to Athene and currently processing
+    private final List<Long> runningAtheneTasks = new ArrayList<>();
+
+    public AtheneService(TextSubmissionService textSubmissionService, TextBlockRepository textBlockRepository, TextBlockService textBlockService,
+            TextClusterRepository textClusterRepository, TextExerciseRepository textExerciseRepository, TextAssessmentQueueService textAssessmentQueueService) {
+        this.textSubmissionService = textSubmissionService;
+        this.textBlockRepository = textBlockRepository;
+        this.textBlockService = textBlockService;
+        this.textClusterRepository = textClusterRepository;
+        this.textExerciseRepository = textExerciseRepository;
+        this.textAssessmentQueueService = textAssessmentQueueService;
+    }
+
+    // region Request/Response DTOs
+    private static class Request {
+
+        public long courseId;
+
+        public String callbackUrl;
+
+        public List<TextSubmission> submissions;
+
+        Request(@NotNull long courseId, @NotNull List<TextSubmission> submissions, @NotNull String callbackUrl) {
+            this.courseId = courseId;
+            this.callbackUrl = callbackUrl;
+            this.submissions = submissionDTOs(submissions);
+        }
+
+        /**
```

Comment:
this comment is not really helpful. Can you explain why you copy those elements?
Why is the method called submissionDTOs if there are no real DTOs involved here?
I would rather call this method `clone` or `copy`